src/reverse_proxy.rs

```diff
@@ -0,0 +1,14 @@
+use std::net::{TcpListener, SocketAddr};
+use crate::errors::*;
+
+pub const DEFAULT_PORT : u16 = 443;
+
+pub fn run(
+    _listen_socket: TcpListener, 
```

Comment:
It's a personal preference, but I typically don't like using underscores in front of variable names unless we need to avoid name conflicts in the same scope.